packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -70,21 +70,41 @@ abstract class BindingBase {
 
   /// The window to which this binding is bound.
   ///
-  /// A number of additional bindings are defined as extensions of [BindingBase],
-  /// e.g., [ServicesBinding], [RendererBinding], and [WidgetsBinding]. Each of
-  /// these bindings define behaviors that interact with a [ui.Window], e.g.,
-  /// [ServicesBinding] registers a [ui.Window.onPlatformMessage] handler, and
-  /// [RendererBinding] registers [ui.Window.onMetricsChanged],
-  /// [ui.Window.onTextScaleFactorChanged], [ui.Window.onSemanticsEnabledChanged],
-  /// and [ui.Window.onSemanticsAction] handlers.
-  ///
-  /// Each of these other bindings could individually access a [Window] statically,
-  /// but that would preclude the ability to test these behaviors with a fake
-  /// window for verification purposes.  Therefore, [BindingBase] exposes this
-  /// [Window] for use by other bindings.  A subclass of [BindingBase], such as
+  /// A number of additional bindings are defined as extensions of
+  /// [BindingBase], e.g., [ServicesBinding], [RendererBinding], and
+  /// [WidgetsBinding]. Each of these bindings define behaviors that interact
+  /// with a [ui.SingletonFlutterWindow].
+  ///
+  /// Each of these other bindings could individually access a
+  /// [ui.SingletonFlutterWindow] statically, but that would preclude the
+  /// ability to test its behaviors with a fake window for verification
+  /// purposes.  Therefore, [BindingBase] exposes this
+  /// [ui.SingletonFlutterWindow] for use by other bindings.  A subclass of
+  /// [BindingBase], such as [TestWidgetsFlutterBinding], can override this
+  /// accessor to return a different [ui.SingletonFlutterWindow] implementation,
+  /// such as a [TestWindow].
+  ui.SingletonFlutterWindow get window => ui.window;
```

Comment:
Should we add a note here that certain things are better accessed via the platformDispatcher instead of via this property?

Reply:
Is there ever a reason for new code to use this property? It seems like the doc comment here should have a little more details around the expected usage of this property.

Reply:
While there really isn't a great reason for new code to use this property, it could be simpler for apps which only have one window. If I could, I'd just remove this object if I didn't think it was massively breaking.

I'll see what I can do about giving better documentation.

Reply:
I'm a little wary of putting too much documentation in here that sounds like we have multi-window support already. I started to write this, and wanted to add things like "For applications with more than one window, prefer using platformDispatcher...", which is really misleading right now, since we don't yet have multi-window support.